File: laboratories/ppo/exercises/lab05/java/Main.java
import lab05.Car;
import lab05.Truck;

public class Main {
  public static void main(String[] args) {
	Car[] vehicles = {
		new Car("Volkswagen Passat"),
		new Truck("Volkswagen Transporter"),
	};

	for(Car car : vehicles) {
		System.out.println(car.getName() + ": " + getPayment(car) + "zł");
	}
  }

  protected static double getPayment(Car car) {
  	return switch(car) {
  		case Truck t -> 5.0;
  		default -> 4.0;
  	};
  }
}

File: laboratories/ppo/exercises/lab05/python/main.py
from src.Car import Car
from src.Truck import Truck


def get_payment(car):
    if isinstance(car, Truck):
        return 5.0
    else:
        return 4.0


vehicles = [
    Car("Volkswagen Passat"),
    Truck("Volkswagen Transporter"),
]

for vehicle in vehicles:
    print(f"{vehicle.get_name()}: {get_payment(vehicle)}zł")
